src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewService.kt

```diff
@@ -175,16 +155,7 @@ class PrisonerIepLevelReviewService(
       throw NoDataFoundException(bookingId)
     }
 
-    prisonerIepLevelRepository.delete(prisonerIepLevel)
-    nextReviewDateUpdaterService.update(bookingId)
-
-    // If the deleted record had `current=true`, the latest IEP review becomes current
-    if (prisonerIepLevel.current) {
-      // The deleted record was current, set new current to the latest IEP review
-      prisonerIepLevelRepository.findFirstByBookingIdOrderByReviewTimeDesc(bookingId)?.run {
-        prisonerIepLevelRepository.save(this.copy(current = true))
-      }
-    }
```

Comment:
Isn't there a way to wrap this in a transaction without moving it out? e.g. 

```Kotlin
@Transactional
{
  // something
  // something
}
// outside transaction
```

I didn't find anything useful.

Reply:
No has to be in a new interface for proxying